### PR TITLE
perf(ext/event): optimize Event constructor

### PR DESCRIPTION
**Pull request by marcosc90**
This PR optimizes `Event` constructor

- ~Added a fast path for empty `eventInitDict`~ Removed `EventInit` dictionary converter
- Don't make `isTrusted` a [LegacyUnforgeable](https://webidl.spec.whatwg.org/#LegacyUnforgeable) property. Doing so makes it non-spec compliant but calling `Object/Reflect.defineProperty` on the constructor is a big bottleneck. Node did the same a few months ago https://github.com/nodejs/node/pull/46974. In my opinion, the performance gains are worth deviating from the spec for a browser-related property.

**This PR**

```
cpu: 13th Gen Intel(R) Core(TM) i9-13900H
runtime: deno 1.36.1 (x86_64-unknown-linux-gnu)

benchmark                      time (avg)        iter/s             (min … max)       p75       p99      p995
------------------------------------------------------------------------------- -----------------------------
event constructor no init      36.69 ns/iter  27,257,504.6   (33.36 ns … 42.45 ns)  37.71 ns  39.61 ns  40.07 ns
event constructor               36.7 ns/iter  27,246,776.6   (33.35 ns … 56.03 ns)  37.73 ns  40.14 ns  41.74 ns
```

**main**

```
cpu: 13th Gen Intel(R) Core(TM) i9-13900H
runtime: deno 1.36.1 (x86_64-unknown-linux-gnu)

benchmark                      time (avg)        iter/s             (min … max)       p75       p99      p995
------------------------------------------------------------------------------- -----------------------------
event constructor no init     380.48 ns/iter   2,628,275.8 (366.66 ns … 399.39 ns) 384.58 ns 398.27 ns 399.39 ns
event constructor             480.33 ns/iter   2,081,882.6 (466.67 ns … 503.47 ns) 484.27 ns 501.28 ns 503.47 ns
```

```js
Deno.bench("event constructor no init", () => {
  const event = new Event("foo");
});

Deno.bench("event constructor", () => {
  const event = new Event("foo", { bubbles: true, cancelable: false });
});
```

towards https://github.com/denoland/deno/issues/20167

### CODE REVIEW

**Comment by marcosc90** [comment]
If we want to adhere to the specification, I will remove the changes to `isTrusted`, but this means that the 'Event' constructor will continue to be relatively slow.

File: tools/wpt/expectation.json [hunk 1]
```diff
@@ -2257,6 +2257,7 @@
     },
     "idlharness-shadowrealm.window.html": false,
     "idlharness.any.worker.html": [
+      "Event interface object length",
```
Comment (lucacasonato) [inline tools/wpt/expectation.json:2260]: Because of the default argument removal

**Review by lucacasonato — COMMENTED** [review]
Looks good

File: tools/wpt/expectation.json (hunk 1 above)
Comment (marcosc90) [reply tools/wpt/expectation.json]: Yeah, you're right, missed that, thought it failed because of isTarget

**Comment by marcosc90** [comment]
After getting more familiar with the spec, we can safely remove `EventInit` dictionary converter. Now calling `new Event` with or without `eventInitDict` yield

```
cpu: 13th Gen Intel(R) Core(TM) i9-13900H
runtime: deno 1.36.1 (x86_64-unknown-linux-gnu)

benchmark              time (avg)        iter/s             (min … max)       p75       p99      p995
------------------------------------------------------------------------------- -----------------------------
event constructor               36.7 ns/iter  27,246,776.6   (33.35 ns … 56.03 ns)  37.73 ns  40.14 ns  41.74 ns
event constructor no init      36.69 ns/iter  27,257,504.6   (33.36 ns … 42.45 ns)  37.71 ns  39.61 ns  40.07 ns
```

from ~2m iter/s to ~27m iter/s

**Comment by bartlomieju** [comment]
Amazing results @marcosc90!

**Comment by mmastrac** [comment]
Amazing result!

File: ext/web/02_event.js [hunk 2]
```diff
@@ -435,6 +412,13 @@ class Event {
   }
 }
 
+// Not spec compliant. The spec defines it as [LegacyUnforgeable]
+// but doing so has a big performance hit
+ReflectDefineProperty(Event.prototype, "isTrusted", {
```
Comment (littledivy) [inline ext/web/02_event.js:417]: LGTM! Nice

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks Marcos!

**Pull request merged**